app/routes/helper_methods.py

```diff
@@ -0,0 +1,38 @@
+from flask import abort,make_response
+from sqlalchemy import asc, desc
+from ..db import db
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except ValueError:
+        response = {"message": f"{cls.__name__} id {model_id} is invalid"}
+        abort(make_response(response, 400))
+
+    query = db.select(cls).where(cls.id == model_id)
+    model = db.session.scalar(query)
+
+    if not model:
+        response = {"message": f"{cls.__name__} with id {model_id} not found"}
+        abort(make_response(response, 404))
+
+    return model
+
+def helper_model_from_dict(cls, request_body, required_fields = ["title", "description"]):
+    for field in required_fields:
+        if field not in request_body:
+            abort(make_response({"details": "Invalid data"}, 400))
```

Comment:
It would be nice if the error response was a little more specific so the client can know why their request body was incorrect and can fix up their request when they send it again. Maybe let the client know which field was incorrect.

```suggestion
        if field not in request_body:
            abort(make_response({"details": f"Invalid data for field: {field}"}, 400))
```